lib/streamlit/runtime/state/session_state_proxy.py

```diff
@@ -42,16 +44,22 @@ def get_session_state() -> SafeSessionState:
 
     ctx = get_script_run_ctx()
 
-    # If there is no script run context because the script is run bare, have
-    # session state act as an always empty dictionary, and print a warning.
+    # If there is no script run context because the script is run bare, we
+    # use a global mock session state version to allow bare script execution (via python script.py)
     if ctx is None:
         if not _state_use_warning_already_displayed:
             _state_use_warning_already_displayed = True
             if not runtime.exists():
                 _LOGGER.warning(
                     "Session state does not function when running a script without `streamlit run`"
                 )
```

Comment:
Is this warning still accurate with the added mock_session state?

Reply:
Yep, to make it visible to the user that the session state will not fully work as expected (if a user executes this not via `streamlit run` for whatever reason. The mocking will just make the bare execution work mostly fine.